### DISCUSSION TITLE
How to exclude example folder when generate changelog

### DISCUSSION THREAD

**Discussion by iTEEECH** [invertase/melos#302] [Q&A] [answered]
Hi

I want to implement the command to generate a changelog at each release. So far so good but I have trouble understanding how the command works. 

Let me explain. I type the command to increase the version manually and generate the changelog file with the different commits. But when it generates the changelog, it also generates me a changelog in the package and in its example while I put "packages/**/example" on ignore. 

changelog root : 

#### `design_system` - `v0.0.1+1`

 - **FEAT**: test. ([xxx](https://github.com/....))

#### `design_system_example` - `v0.0.1+1`

 - **FEAT**: test. ([xxx](https://github.com/....))

changelog example project : 

## 0.0.1+1

 - **FEAT**: test. ([xxxx](https://github.com/....))

And how does Melos distinguish between the different commits in each project when I have several ?

Thanks

**Comment by Ehesp** [comment]
Have you tried adding an ignore flag using the package name? E.g. `--ignore="*_example*"`

**Reply by iTEEECH** [reply]
Yes I tried.

My command : 

` melos version --all --manual-version=xxx_design_system:0.0.1+1 --ignore="*_example*" -c`

result : 

```
Package Name                     Current Version   Updated Version   Update Reason
xxx_design_system           0.0.1+1           0.0.1+2         manual versioning
xxx_design_system_example   0.0.1+1           0.0.1+2           dependency was updated
```

even when I do a Melos bootstrap it detects 2 packages.

**Comment by blaugold** [comment, marked as answer]
Private packages are not included in the changelog. A package is private if it's `pubspec.yaml` has no `version` or if `publish_to` is `none`.

> And how does Melos distinguish between the different commits in each project when I have several ?

For each commit, melos checks if files in a particular package have changed. So if a commit touches files in two different packages, both packages are versioned based on the commit.

**Reply by iTEEECH** [reply]
Yes thank you, but in my case my all packages are private.

In both `pubspec.yaml` files the `publish_to` properties are set to `none`. 

I deleted the version in the `pubspec.yaml` of my example, did a `Melos clean` then `Melos boostrap`.

 I can't seem to ignore the example folder.

**Reply by blaugold** [reply]
With the `--all` flag, all packages are included in the versioning process. In this case, the example package depends on the `desing_system` package and is getting versioned because a "dependency was updated". You can use the `--no-dependent-constraints` flag to disable this. The `--exclude` and `--scope` options only apply to the initial set of packages that should be versioned, but not to the packages that are updated as a consequence of a dependency being updated.

**Reply by iTEEECH** [reply]
Thanks ! It's working.

I use this flag `--no-dependent-constraints` when `Melos version`  and `--ignore="*_example" `with `Melos bootstrap `
